src/Dfc.CourseDirectory.WebV2/ProviderOwnershipCache.cs

```diff
@@ -17,13 +17,43 @@ public ProviderOwnershipCache(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher,
             _cache = cache;
         }
 
+        public async Task<Guid?> GetProviderForCourse(Guid courseId)
+        {
+            var cacheKey = GetApprenticeshipCacheKey(courseId);
+
+            if (!_cache.TryGetValue<Guid?>(cacheKey, out var providerId))
+            {
+                // Cosmos document doesn't store ProviderID on the apprenticeship but UKPRN
+                // so we need two lookups here :-/
+
+                var ukprn = await _cosmosDbQueryDispatcher.ExecuteQuery(
+                    new GetProviderUkprnForCourse() { CourseId = courseId });
+
+                if (ukprn.HasValue)
+                {
+                    var provider = await _cosmosDbQueryDispatcher.ExecuteQuery(
+                        new GetProviderByUkprn() { Ukprn = ukprn.Value });
+
+                    providerId = provider.Id;
+                }
+                else
+                {
+                    providerId = null;
+                }
+
+                _cache.Set(cacheKey, providerId);
```

Comment:
Are you sure that you want to `cache` a null `providerId`?

Reply:
There was some reason this was required for the apprenticeship side of things (but can't remember why) so keeping consistent